src/client/cli/formatter/yaml_formatter.cpp

```diff
@@ -65,93 +65,116 @@ std::string mp::YamlFormatter::format(const InfoReply& reply) const
+            instance_node["disks"].push_back(disk_node);
+
+            YAML::Node memory;
+            memory["usage"] = YAML::Null;
+            memory["total"] = YAML::Null;
+            if (!instance_details.memory_usage().empty())
+                memory["usage"] = std::stoll(instance_details.memory_usage());
+            if (!info.memory_total().empty())
+                memory["total"] = std::stoll(info.memory_total());
+
+            instance_node["memory"] = memory;
+
+            instance_node["ipv4"] = YAML::Node(YAML::NodeType::Sequence);
+            for (const auto& ip : instance_details.ipv4())
+                instance_node["ipv4"].push_back(ip);
+
+            YAML::Node mounts;
+            for (const auto& mount : info.mount_info().mount_paths())
+            {
+                YAML::Node mount_node;
+
+                for (const auto& uid_mapping : mount.mount_maps().uid_mappings())
+                {
+                    auto host_uid = uid_mapping.host_id();
+                    auto instance_uid = uid_mapping.instance_id();
+
+                    mount_node["uid_mappings"].push_back(
+                        fmt::format("{}:{}", std::to_string(host_uid),
+                                    (instance_uid == mp::default_id) ? "default" : std::to_string(instance_uid)));
+                }
+                for (const auto& gid_mapping : mount.mount_maps().gid_mappings())
+                {
+                    auto host_gid = gid_mapping.host_id();
+                    auto instance_gid = gid_mapping.instance_id();
+
+                    mount_node["gid_mappings"].push_back(
+                        fmt::format("{}:{}", std::to_string(host_gid),
+                                    (instance_gid == mp::default_id) ? "default" : std::to_string(instance_gid)));
+                }
+
+                mount_node["source_path"] = mount.source_path();
+                mounts[mount.target_path()] = mount_node;
             }
+            instance_node["mounts"] = mounts;
 
-            mount_node["source_path"] = mount.source_path();
-            mounts[mount.target_path()] = mount_node;
+            info_node[info.name()].push_back(instance_node);
         }
-        instance_node["mounts"] = mounts;
-
-        info_node[info.name()].push_back(instance_node);
     }
+    else if (reply.info_contents_case() == mp::InfoReply::kSnapshotOverview)
+    {
+        for (const auto& item : format::sort_snapshots(reply.snapshot_overview().overview()))
+        {
+            const auto snapshot = item.fundamentals();
+            YAML::Node instance_node;
+            YAML::Node snapshot_node;
+
+            snapshot_node["parent"] = YAML::Null;
+            snapshot_node["comment"] = YAML::Null;
+            if (!snapshot.parent().empty())
+                snapshot_node["parent"] = snapshot.parent();
+            if (!snapshot.comment().empty())
+                snapshot_node["comment"] = snapshot.comment();
```

Comment:
Why not use the ternary operator here? It would reduce 6 LoC to 2 here AFAICS (modulo formatting).

Reply:
Doesn't work here because `YAML::Null` and `std::string` are different types.

Reply:
Close, but this it turns into:
```
snapshot_node["parent"] = snapshot.parent().empty() ? YAML::Node(YAML::NodeType::Null) : YAML::Node(snapshot.parent());
```
It's getting longer, but still ends up saving a few LoC :smile:

Reply:
Heh, true. But this seems to work here and avoid the line splitting:

```
snapshot_node["parent"] = snapshot.parent().empty() ? YAML::Node(YAML::Null) : YAML::Node(snapshot.parent());
```

I think just `YAML::Node{}` also creates a null node, but I am not sure.